src/iter.rs

```diff
@@ -0,0 +1,37 @@
+use crate::TwoFloat;
+use num_traits::Zero;
+use std::iter::Sum;
+use std::ops::Add;
+
+impl<T> Sum<T> for TwoFloat
+where
+    Self: Add<T, Output = Self>,
+{
+    fn sum<I>(iter: I) -> Self
+    where
+        I: Iterator<Item = T>,
+    {
+        iter.fold(Self::zero(), <Self>::add)
+    }
+}
+
+#[cfg(test)]
```

Comment:
For consistency with the existing code, could you move the test methods into a submodule?

```rust
#[cfg(test)]
mod tests {
    #[test]
    fn iter_sum_1() {
        // etc.
    }
}
```

Reply:
I've wrapped it into a module and modified the test to run also with `no_std`